### PR TITLE
Add tests & text for entryname

### PR DESCRIPTION
**Pull request by tetron**

### CODE REVIEW

File: CommandLineTool.yml [hunk 1]
```diff
@@ -68,8 +68,15 @@ $graph:
       * Note that only enum and record types can be typedef-ed
       * Added conformance tests for order of operations evaluating `secondaryFiles` on input
         and ensure that input and output secondaryFiles expressions can return a File object.
-      * Escaping in "string interpolation":#String_Interpolation has
+      * Escaping in [string interpolation](#String_Interpolation) has
         been added to the specification along with conformance tests.
+      * It is now possible to have an absolute path in the `entryname`
+        field in
+        [InitialWorkDirRequirement](#InitialWorkDirRequirement) when
+        running in a container.  Together with
```
Comment (mr-c) [inline CommandLineTool.yml:76]:
```suggestion
        running in a mandatory container.  Together with
```

File: CommandLineTool.yml [hunk 2]
```diff
@@ -883,25 +890,46 @@ $graph:
...
-    Define a file or subdirectory that must be placed in the designated output
-    directory prior to executing the command line tool.  May be the result of
-    executing an expression, such as building a configuration file from a
-    template.
+    Define a file or subdirectory that must be staged to a particular
+    place prior to executing the command line tool.  May be the result
+    of executing an expression, such as building a configuration file
+    from a template.
+
+    Usually files are staged within the designated output directory.
```
Comment (mr-c) [inline CommandLineTool.yml:898]:
```suggestion
    Usually files are staged within the [designated output directory](#Runtime_environment).
```

File: CommandLineTool.yml [hunk 3]
```diff
@@ -883,25 +890,46 @@ $graph:
...
+        a File or Directory, the `entryname` field overrides the value
+        of `basename` of the File or Directory object.
 
         * Required when `entry` evaluates to file contents only
         * Optional when `entry` evaluates to a File or Directory object with a `basename`
         * Invalid when `entry` evaluates to an array of File or Directory objects.
 
+        If `entryname` is a relative path, it specifies a name within
+        the designated output directory.  A relative path starting
+        with `../` is an error.
```
Comment (mr-c) [inline CommandLineTool.yml:918]:
```suggestion
        with `../` or that resolves to location above the designated output directory is an error.
```

File: CommandLineTool.yml [hunk 4]
```diff
@@ -883,25 +890,46 @@ $graph:
...
+
+        If `entryname` is an absolute path (starts with a slash `/`)
+        it is an error unless the following conditions are met:
```
Comment (mr-c) [inline CommandLineTool.yml:921]: Both conditions must be met, or a single condition?

I thought we had agreed upon only allowing this for the 1st condition (mandatory container).

File: CommandLineTool.yml [hunk 5]
```diff
@@ -883,25 +890,46 @@ $graph:
...
+
+          * `DockerRequirement` is present in `requirements`
```
Comment (mr-c) [inline CommandLineTool.yml:923]: This should probably be a bit stronger. Would an empty `DockerRequirement` suffice?

File: CommandLineTool.yml (hunk 4 above)
Comment (tetron) [reply CommandLineTool.yml:921]: Well, the second condition is supposed to just be restating the expectation of isolation you get by running in a container or lightweight virtual machine.  I'm trying to include runtimes that behave like containers (proot) for which this feature would  be reasonable.

File: CommandLineTool.yml (hunk 5 above)
Comment (tetron) [reply CommandLineTool.yml:923]: I don't think an empty DockerRequirement is legal?  Perhaps we should assert that, instead?

Comment (mr-c) [reply CommandLineTool.yml:923]: An empty `DockerRequirement` validates correctly, but it doesn't run with a recent cwltool:

```
$ cwltool --validate test.cwl
INFO cwltool 3.0.20200710214758
test.cwl is valid CWL.
$ cwltool test.cwl
INFO cwltool 3.0.20200710214758
ERROR Workflow or tool uses unsupported feature:
Docker is required to run this tool: 'dockerImageId'
```

File: CommandLineTool.yml [hunk 6]
```diff
@@ -883,25 +890,46 @@ $graph:
...
+        with `../` or that resolves to location above the designated output directory is an error.
+
+        If `entryname` is an absolute path (starts with a slash `/`)
+        it is an error unless the following conditions are met:
+
+          * `DockerRequirement` is present in `requirements`
+          * The program is will run inside a container or other
+          virtualized environment where, from the perspective of
+          the program, the root filesystem is not shared with any
+          other user or running program.
```
Comment (mr-c) [inline CommandLineTool.yml:927]:
```suggestion
          * The program is will run inside a software container
          where, from the perspective of the program, the root
          filesystem is not shared with any other user or
          running program.
```

File: CommandLineTool.yml (hunk 5 above)
Comment (mr-c) [reply CommandLineTool.yml:923]: I don't think we have a schema salad notation for "must have at least one of these properties", though that would be useful!

**Pull request merged**